### PR TITLE
Add memcached and new flags to thanos query frontend

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

### CODE REVIEW

File: environments/base/manifests/query-frontend-cache-service.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/component: store-cache
```
Comment (yeya24) [inline environments/base/manifests/query-frontend-cache-service.yaml:5]: Seems this label is hardcoded in the Memcached library. Does it make sense to make it configurable and change it to `query-frontend-cache`?  I am not sure if this affects anything like dashboard queries.

File: environments/base/manifests/query-frontend-cache-statefulSet.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  labels:
+    app.kubernetes.io/component: store-cache
```
Comment (yeya24) [inline environments/base/manifests/query-frontend-cache-statefulSet.yaml:5]: ditto

**Comment by yeya24** [comment]
Can I get a review of this pr? @kakkoyun @metalmatze 
It blocks the downstream :rofl:

File: environments/base/manifests/query-frontend-cache-service.yaml (hunk 1 above)
Comment (kakkoyun) [reply environments/base/manifests/query-frontend-cache-service.yaml:5]: This should be easily overridden in this repo by changing common labels.

**Review by bwplotka — APPROVED** [review]
@kakkoyun comment to be addressed, otherwise LGTM (:

File: environments/base/manifests/query-frontend-cache-service.yaml (hunk 1 above)
Comment (yeya24) [reply environments/base/manifests/query-frontend-cache-service.yaml]: Done. Please take a look. Addressed in the latest commit https://github.com/observatorium/deployments/pull/360/commits/86df82dd27a8c1b44ef9d8dfbe57ca8b039ad46b

**Review by kakkoyun — APPROVED** [review]
LGTM 🥇

**Pull request merged**